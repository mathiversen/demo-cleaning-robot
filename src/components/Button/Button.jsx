import React from 'react'
import { css } from 'goober'

export default ({ children, onClick }) => {
    const classname = css`
        background: white;
        border: none;
        color: black;
        padding: 0.5rem 1rem;
        border-radius: 0.5rem;
        cursor: pointer;
    `
    return (
        <button onClick={onClick} className={classname}>
            {children}
        </button>
    )
}
