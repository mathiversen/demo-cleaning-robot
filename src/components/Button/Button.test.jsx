import React from 'react'
import Button from './Button'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
    const mockProps = {
        children: [<div key="x">child</div>]
    }
    const tree = renderer.create(<Button {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
})
