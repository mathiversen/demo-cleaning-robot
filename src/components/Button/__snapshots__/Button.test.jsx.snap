// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="go1905345974"
>
  <div>
    child
  </div>
</button>
`;
